packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -247,43 +248,35 @@ export class XInstaller {
   /**
    * Install more plugins to Vue defined by the user.
    *
-   * @param snippet - The snippet configuration.
    * @param bus - The events bus used in the application.
    * @returns The arguments from the plugins installation to be used in Vue's constructor.
    * @internal
    */
-  protected installExtraPlugins(
-    snippet: SnippetConfig,
-    bus: XBus
-  ): Promise<VueConstructorPartialArgument> {
+  protected installExtraPlugins(bus: XBus): Promise<VueConstructorPartialArgument> {
     const vue = this.getVue();
-    return Promise.resolve(this.options.installExtraPlugins?.({ vue, snippet, bus }));
+    return Promise.resolve(
+      this.options.installExtraPlugins?.({ vue, snippet: this.snippetConfig!, bus })
+    );
   }
 
   /**
    * In the case that the `app` parameter is present in the {@link InstallXOptions}, then a new Vue
    * application is created using that app.
    *
    * @param extraPlugins - Vue plugins initialisation data.
-   * @param snippetConfig - Configuration from the client snippet.
    * @returns The Created Vue application or undefined if not created.
    *
    * @internal
    */
-  protected createApp(
-    extraPlugins: VueConstructorPartialArgument,
-    snippetConfig: SnippetConfig
-  ): Vue | undefined {
+  protected createApp(extraPlugins: VueConstructorPartialArgument): Vue | undefined {
     if (this.options.app !== undefined) {
       const vue = this.getVue();
-      this.snippetConfig = vue.observable(snippetConfig);
+      vue.observable(this.snippetConfig);
       return new vue({
         ...extraPlugins,
         ...this.options.vueOptions,
-        provide() {
-          return {
-            snippetConfig
-          };
+        provide: {
+          snippetConfig: this.snippetConfig
```

Comment:
If you end up refactoring the observed snippet config using the returned object instead.

```suggestion
          snippetConfig
```